src/gluonts/nursery/autogluon_tabular.py

```diff
@@ -0,0 +1,118 @@
+from gluonts.model.predictor import Predictor
+
+# Standard library imports
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Iterator,
+    Iterable,
+    Dict,
+)
+
+# Third-party imports
+import numpy as np
+from pandas.tseries.holiday import USFederalHolidayCalendar as calendar
+import pandas as pd
+from autogluon import TabularPrediction as task
+
+# First-party imports
+from gluonts.dataset.common import DataEntry, Dataset
+from gluonts.dataset.util import to_pandas
+from gluonts.model.forecast import Forecast, SampleForecast
+from gluonts.model.predictor import Localizer
+from gluonts.model.estimator import Estimator
+
+if TYPE_CHECKING:  # avoid circular import
+    from gluonts.model.estimator import Estimator
+
+OutputTransform = Callable[[DataEntry, np.ndarray], np.ndarray]
+
+
+def get_prediction_dataframe(series):
+    hour_of_day = series.index.hour
+    month_of_year = series.index.month
+    day_of_week = series.index.dayofweek
+    year_idx = series.index.year
+    target = series.values
+    cal = calendar()
+    holidays = cal.holidays(start=series.index.min(), end=series.index.max())
+    df = pd.DataFrame(
+        zip(
+            year_idx,
+            month_of_year,
+            day_of_week,
+            hour_of_day,
+            series.index.isin(holidays),
+            target,
+        ),
+        columns=[
+            "year_idx",
+            "month_of_year",
+            "day_of_week",
+            "hour_of_day",
+            "holiday",
+            "target",
+        ],
+    )
+    convert_type = {x: "category" for x in df.columns.values[:4]}
+    df = df.astype(convert_type)
+    return df
+
+
+class TabularPredictor(Predictor):
+    def __init__(self, ag_model, freq: str, prediction_length: int,) -> None:
+        self.ag_model = ag_model  # task?
+        self.freq = freq
+        self.prediction_length = prediction_length
+
+    def predict(
+        self, dataset: Iterable[Dict], model=None, as_pandas=False
```

Comment:
What is the purpose of `model` and `as_pandas`? They seem unused, so maybe you can remove them

Reply:
added model and as_pandas to predict as the parameters. I have added them to the predict. It is described in the document: 
modelstr (optional)

> modelstr (optional)
>The name of the model to get predictions from. Defaults to None, which uses the highest scoring model on the validation set. Valid models are listed in this predictor by calling predictor.get_model_names()
> 
> as_pandasbool (optional)
> Whether to return the output as a pandas Series (True) or numpy array (False)

Reply:
Understood. However it doesn't seem like `as_pandas` would be useful, right? It's more of an internal detail, since the output of the Autogluon task's `predict` method will be anyway converted into a `SampleForecast` object. Or am I missing something?

Reply:
Yes. If we just want `SampleForecast`, these parameters can be deleted.